tests/test_client_request.py

```diff
@@ -279,6 +279,37 @@ def test_host_header_ipv6_with_port(make_request: Any) -> None:
     assert req.headers["HOST"] == "[::2]:99"
 
 
+@pytest.mark.parametrize(
+    ("url", "headers", "expected"),
+    (
+        pytest.param("http://localhost.", None, "localhost", id="dot only at the end"),
+        pytest.param("http://python.org.", None, "python.org", id="single dot"),
+        pytest.param(
+            "http://python.org.:99", None, "python.org:99", id="single dot with port"
+        ),
+        pytest.param(
+            "http://python.org...:99",
+            None,
+            "python.org:99",
+            id="multiple dots with port",
+        ),
+        pytest.param(
+            "http://python.org.:99",
+            {"host": "example.com.:99"},
+            "example.com.:99",
```

Comment:
I wonder if this would work well:
```suggestion
            {"host": "tomáš.masaryk.cz.:99"},
            "tomáš.masaryk.cz.:99",
```
Also, let's think of other places where we might need IDN coverage.

Reply:
@webknjaz I've added a different IDNA test case.

Reply:
Thank you 🙏

Marking this thread as resolved.